### PR TITLE
[PyROOT][10454] Prevent memory hogging issue when checking for enums

### PR DESCRIPTION
**Pull request by etejedor**
As reported in issue #10454, calling `ClassInfo_IsEnum()` repeatedly increases the memory consumption of the program. This does not happen with `TEnum::GetEnum()`, which can be used instead.

### CODE REVIEW

**Review by Axel-Naumann — APPROVED** [review]
FYI I checked in the cppyy (incl legacy) history that this isn't roundtripping from `TEnum::GetEnum()` => `gInterpreter->ClassInfo_IsEnum()` and back :-)

**Comment by pcanal** [comment]
@vepadulano Can you check why this seemingly straightforward change is making some test fail?

**Comment by dpiparo** [comment]
Revamp the tests.
Adding in the loop @aaronj0  and @guitargeek , since this is a patch to Cppyy.

**Comment by vepadulano** [comment]
@pcanal I found out at least one of the reasons why this seemingly simple change is triggering unexpected failures. In the tutorial `vo001_AdoptOrOwnMemory.py`, after properly defining all rule of five methods for the class `UponCopyPrinter`, i.e.
```
class UponCopyPrinter {
public:
   UponCopyPrinter() = default;          
   UponCopyPrinter(const UponCopyPrinter &) { std::cout << "Invoking copy c'tor!" << std::endl; }
   UponCopyPrinter(UponCopyPrinter &&) = default;
   UponCopyPrinter &operator=(const UponCopyPrinter &) = default;
   UponCopyPrinter &operator=(UponCopyPrinter &&) = default;
};
```

Then with the changes of this PR I see the following error
```
Traceback (most recent call last):
  File "/home/vpadulan/programs/rootproject/rootbuild/pythonizations-tests/tutorials/vecops/vo001_AdoptOrOwnMemory.py", line 46, in <module>
    v2 = RVec_UponCopyPrinter(v.data(), v.size())
                              ^^^^^^
AttributeError: 'RVec<UponCopyPrinter>' object has no attribute 'data'
```

The reason why this happens is that during the construction of the Python scope proxy, there is a traversal of all base classes of a class. Now, with current master, I see this order of `IsEnum` calls happening (simple printouts added to code)
```
For class UponCopyPrinter IsEnum result is false
For class ROOT::VecOps::RVec<UponCopyPrinter> IsEnum result is false
For class ROOT::VecOps::RVecN<UponCopyPrinter,48> IsEnum result is false
```
Whereas with the changes of this PR I see
```
For class UponCopyPrinter IsEnum result is false
For class ROOT::VecOps::RVec<UponCopyPrinter> IsEnum result is false
For class UponCopyPrinter IsEnum result is false
```
i.e. there is a missing `ROOT::VecOps::RVecN<UponCopyPrinter,48>`. This is due to the fact that when cppyy is trying to build the list of base classes at https://github.com/root-project/root/blob/790fdfa8a9310787d59e050155c4a125d0369e5a/bindings/pyroot/cppyy/CPyCppyy/src/ProxyWrappers.cxx#L688, it asks the class to report on the number of base classes at https://github.com/root-project/root/blob/790fdfa8a9310787d59e050155c4a125d0369e5a/bindings/pyroot/cppyy/CPyCppyy/src/ProxyWrappers.cxx#L415. This in turn calls the function `Cppyy::GetNumBases` defined at https://github.com/root-project/root/blob/790fdfa8a9310787d59e050155c4a125d0369e5a/bindings/pyroot/cppyy/cppyy-backend/clingwrapper/src/clingwrapper.cxx#L1294. The number differs between master and this patch. In master, the class correctly can see its base, reporting 1, whereas with these changes the class reports 0 base classes

## TL;DR

`Cppyy::GetNumBases(19)`, where `19=="ROOT::VecOps::RVec<UponCopyPrinter>"` returns `1` in master and `0` with the simple change of this PR. Seemingly the use of `TEnum::GetEnum` in turn prevents proper population of information about the class hierarchy.

Also FYI @aaronj0 @guitargeek
